addons/mrp/report/mrp_report_bom_structure.py

```diff
@@ -293,14 +299,22 @@ def _get_bom_data(self, bom, warehouse, product=False, line_qty=False, bom_line=
         bom_report_line.update(availabilities)
 
         if level == 0:
-            if bom_report_line['producible_qty'] > 0:
-                bom_report_line['status'] = _("%(qty)s Ready To Produce", qty=bom_report_line['producible_qty'])
-            else:
-                bom_report_line['status'] = _("No Ready To Produce")
-        elif missing_qty := max(bom_report_line['quantity'] - bom_report_line['quantity_available'], 0):
-            missing_qty = float_repr(missing_qty, self.env['decimal.precision'].precision_get('Product Unit'))
-            route_name = bom_report_line['route_name'] or _('Order')
-            bom_report_line['status'] = _("%(qty)s To %(route)s", qty=missing_qty, route=route_name)
+            # Gives a unique key for the first line that indicates if product is ready for production right now.
```

Comment:
I think you can move this whole block at the end of the method where the old `if level == 0` was.
This way, you don't need to rewrite `bom_report_line['bom_cost']` later on if the bom contains operations. And it's easier to understand if *all* `status` definition are in a single place.